common/components/common/projects/ProjectVolunteerModal.jsx

```diff
@@ -0,0 +1,165 @@
+// @flow
+
+import React from 'react';
+import metrics from "../../utils/metrics.js";
+import {Modal, Button, ControlLabel, FormControl, FormGroup} from 'react-bootstrap';
+import ProjectAPIUtils from '../../utils/ProjectAPIUtils.js'
+import CurrentUser from "../../utils/CurrentUser.js";
+import ConfirmationModal from '../../common/confirmation/ConfirmationModal.jsx';
+import TagCategory from "../tags/TagCategory.jsx";
+import TagSelector from "../tags/TagSelector.jsx";
+import {TagDefinition} from "../../utils/ProjectAPIUtils.js";
+import {SelectOption} from "../../types/SelectOption.jsx";
+import Select from 'react-select'
+import moment from 'moment';
+
+
+type Props = {|
+  projectId: number,
+  showModal: boolean,
+  handleClose: () => void
+|};
+type State = {|
+  showModal: boolean,
+  isSending: boolean,
+  message: string,
+  daysToVolunteerForOption: SelectOption,
+  roleTag: TagDefinition,
+  showConfirmationModal: boolean
+|};
+
+const volunteerPeriodsInDays: $ReadOnlyArray<SelectOption> = [
+  ["Less than 1 week",7],
+  ["1 week - 1 month",30],
+  ["1 month - 3 months",90],
+  ["3 months - 6 months",180],
+  ["6 months - 1 year",365]
```

Comment:
Maybe account for all possible durations by adding an open-ended "greater than a year" option?

Reply:
On the database side we're storing the end date, so the question in that case would be what to make the end date.  I think we can revisit this if we find out people want to commit to more than a year.